modules/minileven/minileven.php

```diff
@@ -117,7 +117,9 @@ function jetpack_mobile_template( $theme ) {
 }
 
 function jetpack_mobile_available() {
-	echo '<div class="jetpack-mobile-link" style="text-align:center;margin:10px 0;"><a href="'. home_url( '?ak_action=accept_mobile' ) . '">' . __( 'View Mobile Site', 'jetpack' ) . '</a></div>';
+	global $wp;
+	$current_url =  trailingslashit( home_url( add_query_arg( array(), $wp->request ) ) );
+	echo '<div class="jetpack-mobile-link" style="text-align:center;margin:10px 0;"><a href="'. $current_url . '?ak_action=accept_mobile">' . __( 'View Mobile Site', 'jetpack' ) . '</a></div>';
```

Comment:
What is your rationale behind that change? It seems unrelated to the previous change, so it may be best to include this in a different pull request. 

I would also recommend, since you are using `add_query_arg` here, to use it to add the `ak_action` param.

Reply:
Sure, I am considering doing so. I do so just because ` modules/minileven/theme/pub/minileven/footer.php` did the same way, so it is consistent. However, ` modules/minileven/theme/pub/minileven/footer.php` is doing things wrongly already.
I am testing if `$current_url =  home_url( add_query( array( 'ak_action' => 'accept_mobile' ), add_query_arg( $_GET, $wp->request ) ) );
` will work.

Reply:
Thank you!

Reply:
I have removed the second change. Could you review the pull request and merge it? I will request another soon.

Reply:
Oh, I just remember why I put 2nd change here. Before I fix this bug, the link for web and full is consistent mostly. That is Page A => click web => Page A => click full => Page A. After my update, it becomes Page A => click web => home => click full => home. So should I
1) keep the second change which is buggy but consistent;
2) URL for both web an full;
3) just leave it as it is and create another PR?

Reply:
It may be worth reconsidering your patch. We do need `HTTP_REFERER` in some cases, as this shows, for example when navigating within the site (instead of coming from a different site).

Reply:
I added another commit (i.e. I chose option 2) which solves this issue. Now both navigating within the site and coming from a different site should work.
Anyway, I don't think use referer is a good idea.

Reply:
Noting that the 2nd commit also fix the bug with URL like https://blog.example.com/?p=1.
I have apply the patch to my blog. If you want to do some test you can try it [here](blog.boleyn.su).